src/channel/emotes.js

```diff
@@ -133,6 +144,38 @@ EmoteModule.prototype.sendEmotes = function (users) {
     });
 };
 
+EmoteModule.prototype.handleRenameEmote = function (user, data) {
+    if (typeof data !== "object") {
+        return;
+    }
+
+    if (!this.channel.modules.permissions.canEditEmotes(user)) {
+        return;
+    }
+
+    var f = validateEmote(data);
+    if (!f) {
+        var message = "Unable to rename emote '" + JSON.stringify(data) + "'.  " +
+                "Please contact an administrator for assistance.";
+        if (!data.image || !data.name) {
+            message = "Emote names and images must not be blank.";
+        }
+
+        user.socket.emit("errorMsg", {
+            msg: message,
+            alert: true
+        });
+        return;
+    }
+
+    var chan = this.channel;
+    chan.broadcastAll("renameEmote", f);
+    chan.logger.log(`[mod] ${user.getName()} renamed emote: ${f.old} -> ${f.name}`);
+
+    // Doing this after the broadcast because this function deletes the "old" prop
```

Comment:
Prefer explicitly cloning the object instead of mutating it.  Immutable objects are much easier to reason about.